src/main/java/de/tum/in/www1/artemis/web/rest/DataExportResource.java

```diff
@@ -113,7 +125,56 @@ private boolean currentlyLoggedInUserIsOwnerOfDataExport(DataExport dataExport)
             return false;
         }
         else {
-            return dataExport.getUser().getLogin().equals(SecurityUtils.getCurrentUserLogin().orElseThrow());
+            return dataExport.getUser().getLogin().equals(userRepository.getUser().getLogin());
         }
     }
+
+    /**
+     * Check if the user can request a data export
+     *
+     * @return true if the user can request a data export, false otherwise
+     */
+    @GetMapping("data-exports/can-request")
+    @EnforceAtLeastStudent
+    public boolean canRequestExport() {
+        return canRequestDataExport();
+    }
+
+    /**
+     * Check if the user can download any data export
+     *
+     * @return a data export DTO with the id of the export that can be downloaded or a DTO with a id of null if no export can be downloaded
+     */
+    @GetMapping("data-exports/can-download")
+    @EnforceAtLeastStudent
+    public DataExportDTO canDownloadAnyExport() {
+        return dataExportService.canDownloadAnyDataExport();
+    }
+
+    /**
+     * Check if the user can download a specific data export
+     *
+     * @param dataExportId the id of the data export that should be checked
+     * @return true if the user can download the data export, false otherwise
+     */
+    @GetMapping("data-exports/{dataExportId}/can-download")
+    @EnforceAtLeastStudent
+    public boolean canDownloadSpecificExport(@PathVariable long dataExportId) {
+        return canDownloadSpecificDataExport(dataExportId);
+    }
+
+    /**
+     * Checks if the data export with the given id can be downloaded.
+     *
+     * @param dataExportId the id of the data export to check
+     * @return true if the data export can be downloaded, false otherwise
+     * @throws de.tum.in.www1.artemis.web.rest.errors.EntityNotFoundException  if the data export or the user could not be found
+     * @throws de.tum.in.www1.artemis.web.rest.errors.AccessForbiddenException if the user is not allowed to download the data export
+     */
+    private boolean canDownloadSpecificDataExport(long dataExportId) {
+        var dataExport = dataExportRepository.findByIdElseThrow(dataExportId);
+        currentlyLoggedInUserIsOwnerOfDataExportElseThrow(dataExport);
```

Comment:
This will throw an exception (error response code) when can-download is called with a report the caller has no access to. Is this correct? Should this not rather just return false for this endpoint? Same for the case where the report does not exist?

Reply:
no, the exception is intended because one has to differentiate between can download a specific data export (this casee) where I explicitly go to a page to download a specific data export that might not belong to me. In this case, the exception and 403 error code is the right choice IMO.
And I also prefer the 404 because it's more explicit and just returning a 200 and false if the id doesn't exist
The other case is a check whether there is any data export that is in a downloadable state where we don't throw because this check happens whenever you navigate to the data export page and would lead to a lot of forbidden errors that shouldn't occur because there's nothing wrong if no data export exists that is downloadable.